### PR TITLE
Add Support for Azurite in Storage Live Tests

### PR DESCRIPTION
**Pull request by alzimmermsft**
Adds a pre-run step to install Azurite and run it when running the Storage live tests. NPM install was chosen to install Azurite as there were difficulties with having the docker instance running on the testing VM to use the correct OS, the NPM installation also allows for more flexibility in our control of the run.

### CODE REVIEW

**Comment by danieljurek** [comment]
In the case of cosmos we run the emulator during CI builds (Java: https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/stages/cosmos-sdk-client.yml#L27, Python: https://github.com/Azure/azure-sdk-for-python/blob/master/eng/pipelines/templates/stages/cosmos-sdk-client.yml#L31) 

Is the goal here to test only nightly? If so then these changes belong in tests.yml as you have them.

File: sdk/storage/tests.yml [hunk 1]
```diff
@@ -23,3 +23,8 @@ jobs:
         AZURE_STORAGE_QUEUE_ENDPOINT: $(java-storage-test-queue-endpoint)
         AZURE_STORAGE_QUEUE_CONNECTION_STRING: $(java-storage-test-queue-connection-string)
         AZURE_STORAGE_BLOB_CONNECTION_STRING: $(java-storage-test-blob-connection-string)
+      PreRunSteps:
```
Comment (danieljurek) [inline sdk/storage/tests.yml]: It looks like we're not setting any environment variables in this change that will point the tests to Azurite.

**Comment by alzimmermsft** [comment]
> 
> 
> In the case of cosmos we run the emulator during CI builds (Java: https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/stages/cosmos-sdk-client.yml#L27, Python: https://github.com/Azure/azure-sdk-for-python/blob/master/eng/pipelines/templates/stages/cosmos-sdk-client.yml#L31)
> 
> Is the goal here to test only nightly? If so then these changes belong in tests.yml as you have them.

Correct, Azurite only needs to be started during the nightly test run as the tests against it should have playback records which are used during PR validation.

File: sdk/storage/tests.yml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/tests.yml]: Azurite uses a publicly documented endpoint and "secret", for now I have hard-coded that into my test cases.

Comment (danieljurek) [reply sdk/storage/tests.yml]: I recommend against hardcoding configurable values that might change in the future. Ultimately that decision rests with the developer making the change.

**Comment by alzimmermsft** [comment]
Azurite tests are passing, the live tests are still timing out and need further investigation.

**Pull request merged**